### PR TITLE
films list by Yevhen Sahach

### PR DESCRIPTION
**Pull request by eugeneSahach**
https://yevhen-sahach.github.io/react_films-list-use-raw-redux/

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,37 +1,40 @@
...
-    filmsList: films,
     searchWord: '',
   };
 
+  unsubscribe = null;
+
   componentDidMount() {
     this.searchFilm('spider');
+    this.unsubscribe = store
+      .subscribe(() => this.setState());
```
Comment (YuriiHoliuk) [inline src/App.js]: Why do you use empty setState?

File: src/App.js [hunk 2]
```diff
@@ -1,37 +1,40 @@
...
-        ...prevState.filmsList,
-        {
-          id: prevState.filmsList[prevState.filmsList.length - 1].id + 1,
-          ...newFilm,
-        },
-      ],
+    const { films } = store.getState();
+    store.dispatch(addNewFilm({
+      ...newFilm,
+      id: films[films.length - 1].id + 1,
```
Comment (YuriiHoliuk) [inline src/App.js]: better to create id in reducer or action

File: src/components/FilmDetails/FilmDetails.js [hunk 3]
```diff
@@ -1,15 +1,23 @@
 /* eslint-disable max-len */
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
+import { store } from '../../store/index';
 
 export class FilmDetails extends Component {
+  state = {
+    film: store.getState().films
+      .find(f => String(f.id) === this.props.match.params.id),
```
Comment (YuriiHoliuk) [inline src/components/FilmDetails/FilmDetails.js:9]: What if id will change?

File: src/App.js (hunk 1 above)
Comment (YuriiHoliuk) [reply src/App.js]: Why do you need subscribe here?

**Review by YuriiHoliuk — CHANGES_REQUESTED** [review]

File: src/components/FilmDetails/FilmDetails.js (hunk 3 above)
Comment (eugeneSahach) [reply src/components/FilmDetails/FilmDetails.js:9]: everything will be fine because we have dynamic link if id changes link changes as well

**Comment by irenhh** [comment]
search doesn't work

**Review by irenhh — APPROVED** [review]

**Pull request closed without merging**